xbmc/network/upnp/UPnPServer.cpp

```diff
@@ -1193,9 +1198,10 @@ CUPnPServer::ServeFile(const NPT_HttpRequest&              request,
         return NPT_SUCCESS;
     }
 
-    if(URIUtils::IsURL((const char*)file_path))
+    if (URIUtils::IsURL(static_cast<const char*>(file_path)))
     {
-      std::string disp = "inline; filename=\"" + URIUtils::GetFileName((const char*)file_path) + "\"";
+      CURL url(static_cast<const char*>(file_path));
+      std::string disp = "inline; filename=\"" + URIUtils::GetFileName(url.GetHostName()) + "\"";
```

Comment:
Are we sure that all paths are `image://` paths here because this only makes sense for those paths right?

Reply:
The artwork is wrapped in an image::// url with GetWrappedImageURL(), the actual media files (movie, music) are not.
As a result, this block of code is executed only for artwork because URIUtils::IsURL() is evaluated to false for the rest of the files served. So yes to first part of your question.

However I don't think it makes sense to include a content-disposition header in the first place, unless some uPnP clients require it for compatibility.
That is unlikely though, I tried a couple other well-known uPnP servers and they do not include that header at all (Serviio, Twonky).

Reply:
Ah right but theoretically there could be items in the database which come from FTP / HTTP which would also be considered URLs bei `URIUtils::IsURL()` and for those `CURL::GetHostName()` would not be the proper way to get the filename. Maybe we should extend `URIUtils::GetFileName()` to properly handle `image://` URLs instead?